clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/deploy/ops/CreateServerGroupAtomicOperation.java

```diff
@@ -806,33 +847,37 @@ private String getServerGroupName(Service service) {
   }
 
   private Collection<KeyValuePair> setSpinnakerEnvVars(
-      Collection<KeyValuePair> targetEnv, String newServerGroupName) {
+      Collection<KeyValuePair> targetEnv, EcsServerGroupName newServerGroupName) {
+
+    Moniker m = newServerGroupName.getMoniker();
 
-    targetEnv.add(new KeyValuePair().withName("SERVER_GROUP").withValue(newServerGroupName));
-    targetEnv.add(new KeyValuePair().withName("CLOUD_STACK").withValue(description.getStack()));
     targetEnv.add(
-        new KeyValuePair().withName("CLOUD_DETAIL").withValue(description.getFreeFormDetails()));
+        new KeyValuePair().withName("SERVER_GROUP").withValue(newServerGroupName.getServiceName()));
+    targetEnv.add(new KeyValuePair().withName("CLOUD_STACK").withValue(m.getStack()));
+    targetEnv.add(new KeyValuePair().withName("CLOUD_DETAIL").withValue(m.getDetail()));
 
     return targetEnv;
   }
 
   private Map<String, String> setSpinnakerDockerLabels(
-      Map<String, String> targetMap, String newServerGroupName) {
+      Map<String, String> targetMap, EcsServerGroupName newServerGroupName) {
 
     Map<String, String> newLabels = new HashMap<>();
     if (targetMap != null) {
       newLabels.putAll(targetMap);
     }
 
-    if (description.getStack() != null) {
-      newLabels.put(DOCKER_LABEL_KEY_STACK, description.getStack());
+    Moniker m = newServerGroupName.getMoniker();
+
+    if (StringUtils.isNotEmpty(m.getStack())) {
```

Comment:
The existing behaviour was to add the labels regardless of value, same as the environment variables. I'm not sure it's a good idea to do that, but fixing it would be a breaking change.

Reply:
That's a fair point. I think we should fix it, but I agree it's outside the scope of this PR.